iwyu.cc

```diff
@@ -2083,7 +2083,120 @@ class IwyuBaseAstVisitor : public BaseAstVisitor<Derived> {
     // better for IWYU to require the full type info in such situations. It is
     // important not to suggest a full type when it is decidedly unnecessary
     // for computing the correct value.
+    // See CheckUnaryTypeTraitTypeCompleteness function from
+    // clang/lib/Sema/SemaTypeTraits.cpp for the reference on unary type traits.
     switch (expr->getTrait()) {
+      case TypeTrait::UTT_IsCompleteType:
+        // TODO(bolshakov): would it be better to require the type here because
+        // static_assert(__is_complete_type(Type)) is more probable to appear
```

Comment:
On the other hand, doesn't the `__is_complete_type` check signal that you want conditional behavior depending on whether the full type is available or not? Having IWYU make the full type be available (by adding an `#include`) would make the check redundant in that case...?

Reply:
I still cannot imagine such use cases. I think, they would introduce some sort of UB. But I'd prefer don't require the complete type here for now, too. Do you think the comment should be removed?